src/commands/command.ts

```diff
@@ -24,4 +25,15 @@ export interface Command {
   description: string;
   args: ArgDescriptor[];
   run(options: CommandOptions, config: ProjectConfig): Promise<any>;
-}
\ No newline at end of file
+  extraUsageGroups?(config: ProjectConfig): UsageGroup[];
+}
+
+/**
+ * Running a command may reject with a CommandFailure to indicate that there
+ * was an error but it has already been reported to the user, and the CLI
+ * should exit with the given exit code.
+ */
+export class CommandFailure {
```

Comment:
For the case where the linter has successfully printed warnings? Or the build stopped because of warnings (that have already been printed)?